### ISSUE TITLE
Move TRPO/PPO to new package: PolicyOptimization.jl

### ISSUE THREAD

**Issue by mossr** [sisl/POMDPStressTesting.jl#1]
The TRPO and PPO implementations are general enough to be in their own solver package in the POMDPs.jl ecosystem. I've already encapsulated these solvers into the DeepRL module.

Some TODOs:
- [ ] Extract AST specific code and generalize TRPO and PPO solvers
- [ ] Test on different POMDPs.jl example

Among other things to be appended to this list.

**Comment by MaximeBouton** [comment]
Maybe they could be a TRPO.jl and PPO.jl ? or PolicyOptimization.jl with both?

**Comment by mossr** [comment]
Agreed. They share a lot of base code, so I like PolicyOptimization.jl over DeepRL.jl.

**Comment by MaximeBouton** [comment]
cool, then that is kind of consistent with DeepQLearning.jl 
RLInterface will need to be registered to general before you register PolicyOptimization.jl btw
